diff --git a/os2web_borger_dk_articles.module b/os2web_borger_dk_articles.module
index 2100ed6..03e96fd 100644
--- a/os2web_borger_dk_articles.module
+++ b/os2web_borger_dk_articles.module
@@ -558,16 +558,16 @@ function os2web_borger_dk_articles_sync_submit($form, &$form_state) {
  * @param int $nid
  *   The node we are updating.
  */
-function os2web_borger_dk_articles_update_article($nid) {
+function os2web_borger_dk_articles_update_article(&$node) {
   // First we find the external_id, and last_update time for this article
-  $data = db_query('SELECT external_id, last_updated FROM {os2web_borger_dk_articles} WHERE nid = :nid', array(':nid' => $nid))->fetchObject();
+  $data = db_query('SELECT external_id, last_updated FROM {os2web_borger_dk_articles} WHERE nid = :nid', array(':nid' => $node->nid))->fetchObject();
 
   // Then we fetch the article item from the Borger.dk webservice
   $wsdl = variable_get('os2web_borger_dk_webservice', 'https://www.borger.dk/_vti_bin/borger/ArticleExport.svc?wsdl');
   $article = _os2web_borger_dk_articles_GetArticleByID($data->external_id, $wsdl);
 
   // Now we update the node content with the fetched article content
-  _os2web_borger_dk_articles_update_node_content($nid, $article);
+  _os2web_borger_dk_articles_update_node_content($node, $article);
 }
 
 /**
@@ -1115,12 +1115,10 @@ function _borger_dk_articles_delete_queue_callback($data) {
 /**
  * Function _os2web_borger_dk_articles_update_node_content().
  */
-function _os2web_borger_dk_articles_update_node_content($nid, $article) {
+function _os2web_borger_dk_articles_update_node_content(&$node, $article) {
   if (!empty($article['Exceptions']) || !empty($article['error'])) {
     return;
   }
-  // First we load the corresponding node.
-  $node = node_load($nid, NULL, TRUE);
 
   // And we update all the node-fields with the article-values.
   $node->title = $article['title'];
@@ -1172,7 +1170,7 @@ function _os2web_borger_dk_articles_update_node_content($nid, $article) {
   node_save($node);
   db_update('os2web_borger_dk_articles')
     ->fields(array('last_updated' => strtotime($article['last_updated'])))
-    ->condition('nid', $nid, '=')
+    ->condition('nid', $node->nid, '=')
     ->execute();
 }
 
@@ -1758,9 +1756,9 @@ function _os2web_borger_dk_articles_GetMunicipalityList($wsdl = 'https://www.bor
  *
  * @ingroup actions
  */
-function os2web_borger_dk_articles_update_action($node, $context = array()) {
+function os2web_borger_dk_articles_update_action(&$node, $context = array()) {
   if ($node->type == 'borger_dk_article') {
-    os2web_borger_dk_articles_update_article($node->nid);
+    os2web_borger_dk_articles_update_article($node);
     watchdog('action', 'Update %title with content from borger.dk.', array('%title' => $node->title));
   }
 }
